xain_fl/config/schema.py

```diff
@@ -0,0 +1,389 @@
+
+       # This section correspond to the `Config.server` attribute
+       [server]
+
+       # Address to listen on for incoming gRPC connections
+       host = "[::]"
+       # Port to listen on for incoming gRPC connections
+       port = 50051
+
+
+       # This section corresponds to the `Config.ai` attribute
+       [ai]
+
+       # Path to a file containing a numpy ndarray to use a initial model
+       # weights. The path can be absolute or relative. Relative path are
+       # relative to the config file directory.
+       initial_weights = "./test_array.npy"
+
+       # Number of global rounds the model is going to be trained for. This
+       # must be a positive integer.
+       rounds = 1
+
+       # Number of local epochs per round
+       epochs = 1
+
+       # Minimum number of participants to be selected for a round.
+       min_participants = 1
+
+       # Fraction of total clients that participate in a training round. This
+       # must be a float between 0 and 1.
+       fraction_participants = 1.0
+
+       # This section corresponds to the `Config.storage` attribute
+       [storage]
+
+       # URL to the storage service to use
+       endpoint = "http://localhost:9000"
+
+       # Name of the bucket for storing the aggregated models
+       bucket = "aggregated_weights"
+
+       # AWS secret access to use to authenticate to the storage service
+       secret_access_key = "my-secret"
+
+       # AWS access key ID to use to authenticate to the storage service
+       access_key_id = "my-key-id"
+
+    This config file can be loaded and used as follows:
+
+    .. code-block:: python
+
+       from xain_fl.config import Config
+
+       config = Config.load("example_config.toml")
+
+       assert config.server.host == "[::]"
+       assert config.server.port == 50051
+
+       assert config.ai.initial_weights == "./test_array.npy"
+       assert config.ai.rounds == 1
+       assert config.ai.epochs == 1
+       assert config.ai.min_participants == 1
+       assert config.ai.fraction_participants == 1.0
+
+       assert config.storage.endpoint == "http://localhost:9000"
+       assert config.storage.bucket == "aggregated_weights"
+       assert config.storage.secret_access_key == "my-access-key"
+       assert config.storage.access_key_id == "my-key"
+    """
+
+    def __init__(self, ai: NamedTuple, storage: NamedTuple, server: NamedTuple):
+        self.ai = ai
+        self.storage = storage
+        self.server = server
+
+    @classmethod
+    def from_unchecked_dict(cls: Type[T], dictionary: Mapping[str, Any]) -> T:
+        """Check if the given dictionary contains a valid configuration, and
+        if so, create a `Config` instance from it.
+
+        Args:
+
+            dictionary: a dictionary containing the configuration
+        """
+        try:
+            valid_config = CONFIG_SCHEMA.validate(dictionary)
+        except SchemaError as err:
+            raise InvalidConfig(err.code)
```

Comment:
in this case we are raising another error while handling the one we got, which is not our intention i beliebe.
More information how to handle exceptions can be found here: [Talk about exceptions by Mario Corchero](https://www.youtube.com/watch?v=V2fGAv2R5j8)
```suggestion
            raise InvalidConfig(err.code) as err
```